### ISSUE TITLE
[M0] [iface_namingmode] 'dhcp_relay' container not coming up after addition/deletion of vlan under 'alias' naming mode

### ISSUE THREAD

**Issue by sanjair-git** [sonic-net/sonic-buildimage#14276]
#### Description
As part of the test case '_test_iface_namingmode.py/test_show_vlan_config_', under 'SONIC_CLI_IFACE_MODE' as '_alias_', vlan is created during test setup and deleted during teardown.

During 'vlan' creation, when '_SONIC_CLI_IFACE_MODE=alias sudo config vlan add <vlan-id>_' is issued, it is expected that '_dhcp_relay_' service should get restarted. But '_dhcp_relay_' container goes down and never comes back.

In '_non-alias_' mode, the command '_sudo config vlan add/del <vlan-id>_' works fine with '_dhcp_relay_' service restarting successfully without any issues.

#### Steps to reproduce the issue:
1. Configure 'alias' interface naming mode on DUT using '_sudo config interface_naming_mode alias_' command.
2. Create a new vlan on the DUT using 'alias' mode using '_SONIC_CLI_IFACE_MODE=alias sudo config vlan add <vlan-id>_'

#### Describe the results you received:
Following error is displayed during _vlan_ addition or deletion.

"**_Error: Restart service dhcp_relay failed with error 0_**"

```
Mar 15 13:55:11.584768 ixs-7215-pizza4 INFO python[12494]: ansible-command Invoked with _uses_shell=True _raw_params=SONIC_CLI_IFACE_MODE=alias sudo config vlan add 100 warn=True stdin_add_newline=True strip_empty_ends=True argv=None chdir=None executable=None creates=None removes=None stdin=None
Mar 15 13:55:13.184914 ixs-7215-pizza4 INFO systemd[1]: Stopping dhcp_relay container...
Mar 15 13:55:13.207260 ixs-7215-pizza4 INFO systemd-udevd[12522]: ethtool: autonegotiation is unset or enabled, the speed and duplex are not writable.
Mar 15 13:55:13.207722 ixs-7215-pizza4 INFO systemd-udevd[12522]: Using default interface naming scheme 'v247'.
Mar 15 13:55:13.224497 ixs-7215-pizza4 NOTICE swss#orchagent: :- start: performing log rotate
Mar 15 13:55:13.224497 ixs-7215-pizza4 NOTICE swss#orchagent: :- addVlan: Create an empty VLAN Vlan100 vid:100 vlan_oid:260000000008b1
Mar 15 13:55:13.299391 ixs-7215-pizza4 NOTICE root: Fast boot flag: dhcp_relay false.
Mar 15 13:55:13.316573 ixs-7215-pizza4 NOTICE root: Warm boot flag: dhcp_relay false.
Mar 15 13:55:13.334293 ixs-7215-pizza4 NOTICE root: Stopping dhcp_relay service...
Mar 15 13:55:14.188230 ixs-7215-pizza4 DEBUG container: container_stop: BEGIN
Mar 15 13:55:14.192880 ixs-7215-pizza4 DEBUG container: read_data: config:True feature:dhcp_relay fields:[('set_owner', 'local'), ('no_fallback_to_local', False), ('state', 'disabled')] val:['local', False, 'enabled']
Mar 15 13:55:14.195796 ixs-7215-pizza4 DEBUG container: read_data: config:False feature:dhcp_relay fields:[('current_owner', 'none'), ('remote_state', 'none'), ('container_id', '')] val:['none', 'none', '']
Mar 15 13:55:14.198696 ixs-7215-pizza4 DEBUG container: container_stop: dhcp_relay: set_owner:local current_owner:none remote_state:none docker_id:dhcp_relay
Mar 15 13:55:14.385949 ixs-7215-pizza4 INFO dhcp_relay#supervisord 2023-03-15 13:55:14,383 WARN received SIGTERM indicating exit request
Mar 15 13:55:14.386620 ixs-7215-pizza4 INFO dhcp_relay#supervisord 2023-03-15 13:55:14,384 INFO waiting for supervisor-proc-exit-listener, rsyslogd, isc-dhcpv4-relay-Vlan1000, dhcp6relay, dhcpmon-Vlan1000 to die
Mar 15 13:55:14.386620 ixs-7215-pizza4 ALERT dhcp_relay#dhcpmon[195]: Received signal: 'Terminated'
Mar 15 13:55:14.387280 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [            eth0- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.387639 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [  PortChannel106- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.387991 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [  PortChannel105- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.388341 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [  PortChannel103- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.388719 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [  PortChannel101- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.389070 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [      Ethernet47- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.389406 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [      Ethernet46- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.389731 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [        Vlan1000- Current rx/tx] Discover:        13/0, Offer:         3/0, Request:  30856760/0, ACK:         0/0
Mar 15 13:55:14.390060 ixs-7215-pizza4 NOTICE dhcp_relay#dhcpmon[195]: [    Agg-Vlan1000- Current rx/tx] Discover:         3/0, Offer:  30856760/0, Request:         0/0, ACK: 3065728536/0
Mar 15 13:55:14.668581 ixs-7215-pizza4 INFO dhcp_relay#supervisord 2023-03-15 13:55:14,667 INFO stopped: dhcpmon-Vlan1000 (exit status 0)
Mar 15 13:55:15.673590 ixs-7215-pizza4 ALERT dhcp_relay#dhcp6relay: Received signal: 'Terminated'
Mar 15 13:55:15.675637 ixs-7215-pizza4 INFO dhcp_relay#supervisord 2023-03-15 13:55:15,674 INFO stopped: isc-dhcpv4-relay-Vlan1000 (terminated by SIGTERM)
Mar 15 13:55:15.694060 ixs-7215-pizza4 INFO dhcp_relay#supervisord 2023-03-15 13:55:15,692 INFO stopped: dhcp6relay (exit status 0)
Mar 15 13:55:16.869482 ixs-7215-pizza4 INFO dockerd[556]: time="2023-03-15T13:55:16.867889680Z" level=info msg="ignoring event" container=a3445de16ae7e62eba3ca94d5cbe7b772a47bb3039e3cad392c70512c3a074ff module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
Mar 15 13:55:16.874211 ixs-7215-pizza4 INFO containerd[522]: time="2023-03-15T13:55:16.872759040Z" level=info msg="shim disconnected" id=a3445de16ae7e62eba3ca94d5cbe7b772a47bb3039e3cad392c70512c3a074ff
Mar 15 13:55:16.874681 ixs-7215-pizza4 INFO containerd[522]: time="2023-03-15T13:55:16.872934480Z" level=warning msg="cleaning up after shim disconnected" id=a3445de16ae7e62eba3ca94d5cbe7b772a47bb3039e3cad392c70512c3a074ff namespace=moby
Mar 15 13:55:16.875035 ixs-7215-pizza4 INFO containerd[522]: time="2023-03-15T13:55:16.873018840Z" level=info msg="cleaning up dead shim"
Mar 15 13:55:16.913648 ixs-7215-pizza4 INFO containerd[522]: time="2023-03-15T13:55:16.913370640Z" level=warning msg="cleanup warnings time=\"2023-03-15T13:55:16Z\" level=info msg=\"starting signal loop\" namespace=moby pid=12576\n"
Mar 15 13:55:16.934716 ixs-7215-pizza4 INFO systemd[1]: var-lib-docker-overlay2-effd5b9979b8965bb61f171823d6c1c9a1691053708a2f92361f22e994a5db9c-merged.mount: Succeeded.
Mar 15 13:55:16.986122 ixs-7215-pizza4 INFO container: docker cmd: wait for dhcp_relay
Mar 15 13:55:16.994436 ixs-7215-pizza4 INFO container: docker cmd: stop for dhcp_relay
Mar 15 13:55:16.996805 ixs-7215-pizza4 DEBUG container: container_stop: END
Mar 15 13:55:17.117609 ixs-7215-pizza4 NOTICE root: Stopped dhcp_relay service...
Mar 15 13:55:17.129458 ixs-7215-pizza4 INFO systemd[1]: dhcp_relay.service: Succeeded.
Mar 15 13:55:17.131787 ixs-7215-pizza4 INFO systemd[1]: Stopped dhcp_relay container.
```
#### Describe the results you expected:

'_dhcp_relay_' service should come back UP after successful restart during vlan addition or deletion in 'alias' interface naming mode.

#### Output of `show version`:

```
SONiC Software Version: SONiC.20220531.18
Distribution: Debian 11.6
Kernel: 5.10.0-18-2-armmp
Build commit: 8c25164188
Build date: Wed Mar  1 18:49:39 UTC 2023
Built by: cloudtest@23e4d742c000000

Platform: armhf-nokia_ixs7215_52x-r0
HwSKU: Nokia-7215
ASIC: marvell
ASIC Count: 1
Serial Number: NK205210110
Model Number: 3HE16794AARA01
Hardware Revision: 3
Uptime: 18:48:17 up 1 day, 22:09,  2 users,  load average: 1.39, 1.22, 0.82
Date: Thu 16 Mar 2023 18:48:17

```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by Blueve** [comment]
@jcaiMR  @yaqiangz can you check this? seems like the DHCP relay binding is not work as expect with different naming mode

**Comment by yaqiangz** [comment]
Hi @sanjair-git,  this problem has been fixed by: https://github.com/sonic-net/sonic-utilities/pull/2707

**Comment by sanjair-git** [comment]
> Hi @sanjair-git, this problem has been fixed by: [sonic-net/sonic-utilities#2707](https://github.com/sonic-net/sonic-utilities/pull/2707)

@yaqiangz , Thanks for the update

**Comment by yaqiangz** [comment]
Has been fixed by https://github.com/sonic-net/sonic-utilities/pull/2707
